sklearn/utils/__init__.py

```diff
@@ -463,12 +488,37 @@ def _get_column_indices(X, key):
             raise ValueError("A given column is not a column of the dataframe") from e
 
         return column_indices
+
+
+def _get_column_indices_interchange(X_interchange, key, key_dtype):
+    """Same as _get_column_indices but for X with __dataframe__ protocol."""
+    n_columns = X_interchange.num_columns()
+
+    if isinstance(key, (list, tuple)) and not key:
```

Comment:
Why do we return an empty list here?

Reply:
It was to match the behavior of `_get_column_indices` on `main`:

https://github.com/scikit-learn/scikit-learn/blob/5d83a2e1efccd7ea265cf2ca254a26f9f57f30c4/sklearn/utils/__init__.py#L415-L417